src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -83,8 +84,39 @@ public async IAsyncEnumerable<FeatureDefinition> GetAllFeatureDefinitionsAsync()
 
         private FeatureDefinition ReadFeatureDefinition(string featureName)
         {
-            IConfigurationSection configuration = GetFeatureDefinitionSections()
-                                                    .FirstOrDefault(section => section.Key.Equals(featureName, StringComparison.OrdinalIgnoreCase));
+            IConfigurationSection configuration;
+
+            if (!featureName.Contains(ConfigurationPath.KeyDelimiter))
+            {
+                configuration = GetFeatureDefinitionSections()
+                    .FirstOrDefault(section => section.Key.Equals(featureName, StringComparison.OrdinalIgnoreCase));
+            }
+            else
+            {
+                //
+                // Feature names with configuration path delimiters require traversing children sections for resolution
+                IEnumerable<string> sectionNames = featureName.Split(SectionSeparator, StringSplitOptions.None);
+
+                IEnumerable<IConfigurationSection> sections;
+
+                IConfigurationSection section = null;
+
+                foreach (string sectionName in sectionNames)
+                {
+                    sections = section == null ?
+                        GetFeatureDefinitionSections() :
+                        section.GetChildren();
```

Comment:
`section == null` will only be true for the first iteration in the foreach loop. Instead of doing this check inside the loop every time, can we initialize `sections` to `GetFeatureDefinitionSections()` before the loop?